tracer.go

```diff
@@ -187,35 +185,54 @@ func (t *tracerImpl) StartSpanWithOptions(
 	tags := opts.Tags
 
 	// Build the new span. This is the only allocation: We'll return this as
-	// a opentracing.Span.
-	sp := t.getSpan()
-	if opts.Parent == nil {
-		sp.raw.TraceID, sp.raw.SpanID = randomID2()
-		sp.raw.Sampled = t.options.shouldSample(sp.raw.TraceID)
-	} else {
-		pr := opts.Parent.(*spanImpl)
-		sp.raw.TraceID = pr.raw.TraceID
-		sp.raw.SpanID = randomID()
-		if t.options.clientServerSameSpan {
```

Comment:
I did not understand why this conditional needed to be here, so I removed it... seems to me that we should only care about `clientServerSameSpan` for RPCs, and we handle that case just below.